src/Features/Hud/VelocityGraph.cpp

```diff
@@ -0,0 +1,140 @@
+#include "VelocityGraph.hpp"
+
+#include "Event.hpp"
+#include "Modules/Client.hpp"
+#include "Modules/Engine.hpp"
+#include "Modules/Scheme.hpp"
+#include "Modules/Surface.hpp"
+#include "Variable.hpp"
+
+Variable sar_hud_velocitygraph("sar_hud_velocitygraph", "0", "Shows velocity graph.\n");
+Variable sar_hud_velocitygraph_font_index("sar_hud_velocitygraph_font_index", "1", 0, "Font index of velocity graph.\n");
+
+VelocityGraph velocityGraph;
+
+VelocityGraph::VelocityGraph()
+	: Hud(HudType_InGame | HudType_Paused | HudType_Menu, true) {
+}
+bool VelocityGraph::ShouldDraw() {
+	return sar_hud_velocitygraph.GetBool() && Hud::ShouldDraw();
+}
+void VelocityGraph::GatherData(int slot) {
+	auto player = client->GetPlayer(slot + 1);
+
+	if (!player) {
+		if (!data.empty())
+			data.clear();
+
+		return;
+	}
+
+	if (data.size() > 500)
+		data.pop_back();
+
+	velocity_data_t current_data;
+
+	auto vel = client->GetLocalVelocity(player);
+	int speed = vel.Length2D();
+
+	unsigned int groundHandle = *(unsigned int *)((uintptr_t)player + Offsets::C_m_hGroundEntity);
+	bool on_ground = groundHandle != 0xFFFFFFFF;
+
+	current_data.speed = speed;
+	current_data.on_ground = on_ground;
+
+	data.insert(data.begin(), current_data);
+}
+
+static int last_vel = 0;
+static int tick_prev = 0;
+
+static int take_off = 0;
+static int take_off_time = 0;
+
+void VelocityGraph::Paint(int slot) {
+	GatherData(slot);
+
+	auto player = client->GetPlayer(slot + 1);
+
+	if (!player) {
+		if (!data.empty())
+			data.clear();
+
+		return;
+	}
+
+	auto vel = client->GetLocalVelocity(player);
+	int speed = vel.Length2D();
+
+	if (data.size() < 2)
+		return;
+
+	int x, y;
+#ifdef _WIN32
+	engine->GetScreenSize(x, y);
+#else
+	engine->GetScreenSize(nullptr, x, y);
+#endif
+
+	for (auto i = 0ul; i < data.size() - 1; i++) {
+		const auto current = data[i];
+		const auto next = data[i + 1];
+
+		const auto clamped_current_speed = std::clamp(current.speed, 0, 1000);
+		const auto clamped_next_speed = std::clamp(next.speed, 0, 1000);
+
+		const int graph_pos[2] = {
+			x / 2 + 125,
+			y - 175 };
+
+		int current_speed = (clamped_current_speed * 75 / 320);
+		int next_speed = (clamped_next_speed * 75 / 320);
+
+		int alpha = i == 0 ? 0 : 255;
+
+		surface->DrawColoredLine(
+			graph_pos[0] - (i - 1) / 2,
+			graph_pos[1] - current_speed,
+
+			graph_pos[0] - i / 2,
+			graph_pos[1] - next_speed,
+
+			Color(255, 255, 255, alpha)
+		);
+	}
+
+	static bool last_on_ground = false;
```

Comment:
Ditto the slot thing for this static variable; I have no idea why this is static within the function rather than just being another global. Either is fine, but be consistent